### PR TITLE
document use of @eval with @everywhere for broadcast of local variables.

### LINKED ISSUE #9118: @everywhere for broadcasting variables
```
addprocs(1)
foo=1
@everywhere bar=foo
```

results in
```
julia> @everywhere bar=foo
exception on 2: ERROR: foo not defined
 in eval at /home/tan/julia/julia/base/sysimg.jl:7
```

While

```
@spawnat 2 global bar=foo
remotecall_fetch(2, ()->bar)
```
works perfectly fine.

Shouldn't `@everywhere` which runs in the scope of `Main` be a perfectly appropriate method to broadcast variables?

### PR DESCRIPTION
**Pull request by amitmurthy**
closes #9118

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -144,7 +144,19 @@ download
...
+
+    @everywhere bar=1
+
+will define `bar` under module `Main` on all processes.
+
+Unlike `@spawn` and `@spawnat`, `@everywhere` does not capture any local variables. Prefixing
+`@everywhere` with `@eval` allows us to broadcast local variables using interpolation :
+
+    foo = 1
+    @eval @everywhere bar=1
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: where's the interpolation?

Comment (amitmurthy) [inline base/docs/helpdb/Base.jl]: Right! Thanks,

**Comment by amitmurthy** [comment]
@bjarthur, thanks for the suggestion. Have modified the commit entry to mention you.

**Comment by bjarthur** [comment]
you're welcome!

**Comment by amitmurthy** [comment]
Travis error is unrelated. In there are no further objections to the text, will merge in a short while.

**Pull request merged**